internal/handler/user.go

```
@@ -0,0 +1,122 @@
+		return
+	}
+
+	createdUser, err := userRepo.CreateUser(r.Context(), user)
+	if err != nil {
+		logger.Error("failed to create user", "error", err)
+		http.Error(w, "internal server error", http.StatusInternalServerError)
+		return
+	}
+
+	w.WriteHeader(http.StatusCreated)
+	json.NewEncoder(w).Encode(createdUser)
```

golangci-lint (1.62.2)

Error return value of `(*encoding/json.Encoder).Encode` is not checked

(errcheck)